esp/services/ws_workunits/ws_workunitsQuerySets.cpp

```diff
@@ -910,19 +915,48 @@ bool CWsWorkunitsEx::onWUPublishWorkunit(IEspContext &context, IEspWUPublishWork
         updateFlags |= DALI_UPDATEF_SUPERFILES;
     if (req.getAppendCluster())
         updateFlags |= DALI_UPDATEF_APPEND_CLUSTER;
+    if (req.getDfuCopyFiles())
+        updateFlags |= DFU_UPDATEF_COPY;
+    if (req.getDfuOverwrite())
+        updateFlags |= DFU_UPDATEF_OVERWRITE;
 
+    StringBuffer publisherWuid;
     if (!req.getDontCopyFiles())
     {
         QueryFileCopier cpr(target);
-        cpr.init(context, req.getAllowForeignFiles());
+        cpr.init(context, req.getAllowForeignFiles(), queryName);
         cpr.remoteIP.set(daliIP);
         cpr.remotePrefix.set(srcPrefix);
         cpr.srcCluster.set(srcCluster);
         cpr.queryname.set(queryName);
-        cpr.copy(cw, updateFlags);
+        cpr.dfu_queue.set(req.getDfuQueue());
+        cpr.copy(publisherWuid, cw, updateFlags);
 
         if (req.getIncludeFileErrors())
             cpr.gatherFileErrors(resp.getFileErrors());
+        if (!publisherWuid.isEmpty())
+            resp.setDfuPublisherWuid(publisherWuid);
+    }
+
+    if (req.getOnlyCopyFiles() || (req.getStopIfFilesCopied() && !publisherWuid.isEmpty()))
```

Comment:
Code like this appears too many times in this file. Is there a way to common it up?  (I know there is already lots of duplication). It is hard to review, and a pain to change if a problem is found.

Reply:
It happens because we have different interfaces with the same or similar members.  But at least in cases like this one, I can use a template function to common it up, so will do.